barretenberg/cpp/src/barretenberg/bbapi/bbapi_ecc.cpp

```diff
@@ -6,19 +6,46 @@
 
 namespace bb::bbapi {
 
-GrumpkinMul::Response GrumpkinMul::execute(BB_UNUSED BBApiRequest& request) &&
+GrumpkinMul::Response GrumpkinMul::execute(BBApiRequest& request) &&
 {
-    return { point * scalar };
+    if (!point.on_curve()) {
+        BBAPI_ERROR(request, "Input point must be on the curve");
+    }
+    auto result = point * scalar;
+    if (!result.on_curve()) {
```

Comment:
This is and other similar checks are probably not necessary since they follow from the inputs being on the curve